damage_window.py

```diff
@@ -0,0 +1,234 @@
+
+
+class DamageBar(BoundingBoxObject):
+    def __init__(self, x, y, w, h):
+        super(DamageBar, self).__init__(x, y, w, h, (255,0,0))
+
+        self.champion = BoundingBoxObject(
+            self.bbox.x + 2,
+            self.bbox.y,
+            self.bbox.h,
+            self.bbox.h,
+            color=(255,0,255)
+        )
+
+        self.dmgValue = BoundingBoxObject(
+            self.champion.bbox.x + self.champion.bbox.h,
+            self.bbox.y,
+            self.bbox.w - self.champion.bbox.w,
+            self.bbox.h / 2,
+            color=(0,0,255)
+        )
+
+        self.levelStars = BoundingBoxObject(
+            self.champion.bbox.x,
+            self.champion.bbox.y + 0.75 * self.champion.bbox.h,
+            self.champion.bbox.w,
+            self.champion.bbox.h * 0.25,
+            color=(0,255,255)
+        )
+
+    def draw(self, img):
+        super(DamageBar, self).draw(img)
+        self.champion.draw(img)
+        self.dmgValue.draw(img)
+        self.levelStars.draw(img)
+
+
+class DamageWindow(object):
+    def __init__(self, anchor, dim, padding, img=None):
+        self.damage_bars = DamageWindow.GenerateDamageBars(anchor, dim, padding, 10)
+
+        # this section is used to test if window is open
+        test_dim = (0.07 * dim[0], (dim[1] + padding) * 3 - padding)
+        self.open_test_section = BoundingBoxObject(anchor[0] - test_dim[0] - 2, anchor[1], test_dim[0], test_dim[1], (255,255,0))
+        self.open_color = np.array([31, 32, 21])
+        # if image is provided calculate open region color
+        if img is not None:
+            self.open_color = np.array(self.GetAverageColor(self.open_test_section.getImage(img)))
+
+    def isOpen(self, img):
+        '''
+        check if damage window is open
+        returns true if open
+        '''
+        color = np.array(self.GetAverageColor(self.open_test_section.getImage(img)))
+        in_range = np.logical_and(color >= self.open_color - 1.5, color <= self.open_color + 1.5)
+        return np.all(in_range)
+
+    def draw(self, img):
+        for bar in self.damage_bars:
+            bar.draw(img)
+        self.open_test_section.draw(img)
+
+    @staticmethod
+    def GetAverageColor(img):
+        return img.mean(axis=0).mean(axis=0)
+
+    @staticmethod
+    def GenerateDamageBars(anchor, dim, padding, n_bars):
+        x, y = anchor
+        bars = []
+        for i in range(n_bars):
+            bars.append(DamageBar(x, y, dim[0], dim[1]))
+            y += (padding + dim[1])
+        return bars
+
+
+class DamageWindowBuilder(object):
+    NUM_BARS = 10
+    
+    # window width (used to make sure the window is fully expanded)
+    WINDOW_WIDTH = 0
+
+    @staticmethod
+    def InitFromCoods(top_left = [1777, 326], width = 128, height = 36, padding = 9):
+        return DamageWindow(top_left, (width, height), padding)
+
+    @staticmethod
+    def InitFromFrame(img, wait_expansion = False):
```

Comment:
Kinda seems like there are 3 methods in here which seem to use the same img argument. Fair to make this a non static class with that shared argument?

Reply:
Ya that could be done but for now seems like too much work for no real benefit